patches/telescope/0001-Managarm-Patches.patch

```diff
@@ -0,0 +1,73 @@
+From 084d8d40a24a601c463955ac569183cfb085f96a Mon Sep 17 00:00:00 2001
+From: Alexander <electrodeyt@gmail.com>
+Date: Thu, 16 Jun 2022 00:31:32 +0200
+Subject: [PATCH] Managarm Patches
+
+---
+ control.c   | 14 ++++++++------
+ telescope.c |  9 +++++----
+ 2 files changed, 13 insertions(+), 10 deletions(-)
+
+diff --git a/control.c b/control.c
+index b2cba81..dc2cdcd 100644
+--- a/control.c
++++ b/control.c
+@@ -91,12 +91,14 @@ control_init(char *path)
+ 	}
+ 	umask(old_umask);
+ 
+-	if (chmod(path, S_IRUSR|S_IWUSR|S_IRGRP|S_IWGRP) == -1) {
+-		warn("%s: chmod", __func__);
+-		close(fd);
+-		(void)unlink(path);
+-		return (-1);
+-	}
++
++	// patched because chmod == nop
++	//if (chmod(path, S_IRUSR|S_IWUSR|S_IRGRP|S_IWGRP) == -1) {
++	//	warn("%s: chmod", __func__);
++	//	close(fd);
++	//	(void)unlink(path);
++	//	return (-1);
++	//}
+ 
+ 	return (fd);
+ }
+diff --git a/telescope.c b/telescope.c
+index 0a65f41..1991d1b 100644
+--- a/telescope.c
++++ b/telescope.c
+@@ -64,7 +64,7 @@ int			 operating;
+  * "Safe" (or "sandobox") mode.  If enabled, Telescope shouldn't write
+  * anything to the filesystem or execute external programs.
+  */
+-int			safe_mode;
++int			safe_mode = 0;
+ 
+ static struct imsgev	*iev_net;
+ 
+@@ -1047,8 +1047,8 @@ main(int argc, char * const *argv)
+ 
+ 	if (getenv("NO_COLOR") != NULL)
+ 		enable_colors = 0;
+-
+-	while ((ch = getopt_long(argc, argv, opts, longopts, NULL)) != -1) {
++	optind = 0;
++	while ((ch = getopt(argc, argv, opts)) != -1) {
+ 		switch (ch) {
+ 		case 'C':
+ 			exit(ui_print_colors());
+@@ -1128,8 +1128,9 @@ main(int argc, char * const *argv)
+ 		    "telescope already running?");
+ 	}
+ 
++
+ 	/* Start children. */
+-	if (socketpair(AF_UNIX, SOCK_STREAM, PF_UNSPEC, pipe2net) == -1)
++	if (socketpair(AF_UNIX, SOCK_STREAM, 0, pipe2net) == -1)
```

Comment:
Why is this done? If we're missing `PF_UNSPEC`, adding it to mlibc should be simple.

Reply:
We have it, however Managarm code asserts that the protocol (3rd argument) is 0. We should change the Managarm code to allow PF_UNSPEC (which is basically pick what you think is best afaik).